raspberry/bzzz/sensors/anemometer.py

```diff
@@ -1,61 +1,137 @@
 import serial
+import numpy as np
+from threading import Thread, Lock
+import abc
+
+
+class DataProcessor(abc.ABC):
+
+    def __init__(self):
+        pass
+
+    @abc.abstractmethod
+    def process(self, data, cursor=0):
+        pass
+
+
+class AverageFilter(DataProcessor):
+
+    def __init__(self):
+        super().__init__()
+
+    def process(self, data):
+        return np.nanmean(data, axis=0)
+
+
+class MedianFilter(DataProcessor):
+
+    def __init__(self):
+        super().__init__()
+
+    def process(self, data):
+        return np.nanmedian(data, axis=0)
+
 
 class Anemometer:
-    
-    def __init__(self, serial_path='/dev/ttyS0', baud=115200):
-        self.__ser = serial.Serial(serial_path, baud, timeout=1)
-        self.__ser.reset_input_buffer()
-        self.__split_data_float = None
-    
-    def get(self):
+    """
+    Anemometer sensor
+
+    This class is used to interface the anemometer
+    """
+
+    def __init__(self,
+                 serial_path='/dev/ttyS0',
+                 baud=115200,
+                 window_length=3,
+                 data_processor=MedianFilter()):
+        """
+        Create a new instance of Anemometer
+
+        :param serial_path: serial path; defaults to /dev/ttyS0 on RPi
+        :param baud: baud rate of serial communication; defaults to 115200
+        :param window_length: length of window of measurements; default: 2
```

Comment:
Should this say default: 3 instead of 2?

Reply:
Good catch! Fixed.